util/base/src/main/java/jetbrains/jetpad/base/ThrowableHandlers.java

```diff
@@ -81,6 +85,48 @@ private static boolean isInUnitTests(Throwable t) {
     return false;
   }
 
+  static void handleError(Throwable t) {
+
+    if (isClient(t)) {
+      return;
+    }
+
+    Error error = getError(t);
+
+    if (error != null) {
+      throw error;
+    }
+  }
+
+  private static boolean isClient(Throwable t) {
```

Comment:
Previously you asked, "why do we need to distinguish client and server here?". The answer is we want to keep the current client-side logic "as is". Plus, GWT is designed to deal with Throwables, so we don't want to narrow it.